internal/print/settings.go

```diff
@@ -160,6 +168,51 @@ func (s *Settings) Convert() *printsdk.Settings {
 		ShowSensitivity:  s.ShowSensitivity,
 		ShowRequirements: s.ShowRequirements,
 		ShowResources:    s.ShowResources,
+		ShowDataSources:  s.ShowDataSources,
 		ShowType:         s.ShowType,
 	}
 }
+
+// GetSectionsToPrint returns the sections that'll be printed
+func GetSectionsToPrint(settings *Settings, module *terraform.Module, sections *terraform.Module) *terraform.Module {
+	if settings.ShowHeader {
+		sections.Header = module.Header
+	}
+	if settings.ShowFooter {
+		sections.Footer = module.Footer
+	}
+	if settings.ShowInputs {
+		sections.Inputs = module.Inputs
+	}
+	if settings.ShowModuleCalls {
+		sections.ModuleCalls = module.ModuleCalls
+	}
+	if settings.ShowOutputs {
+		sections.Outputs = module.Outputs
+	}
+	if settings.ShowProviders {
+		sections.Providers = module.Providers
+	}
+	if settings.ShowRequirements {
+		sections.Requirements = module.Requirements
+	}
+	if settings.ShowResources || settings.ShowDataSources {
+		sections.Resources = FilterResourcesByMode(settings, module.Resources)
+	}
+
+	return sections
+}
+
+// FilterResourcesByMode returns the managed or data resources defined by the show argument
+func FilterResourcesByMode(settings *Settings, module []*terraform.Resource) []*terraform.Resource {
```

Comment:
`FilterResourcesByMode` doesn't have to be public, it's better to be `filterResourcesByMode` to not giving any impression than it can publicly be used.